src/DiffEqGPU.jl

```diff
@@ -281,7 +292,28 @@ function batch_solve(ensembleprob,alg,ensemblealg::EnsembleArrayAlgorithm,I;kwar
 
     u0 = reduce(hcat,Array(probs[i].u0) for i in 1:length(I))
     p  = reduce(hcat,probs[i].p isa SciMLBase.NullParameters ? probs[i].p : Array(probs[i].p)  for i in 1:length(I))
-    sol, solus = batch_solve_up(ensembleprob,probs,alg,ensemblealg,I,u0,p;kwargs...)
+
+    sol, solus = 
+    if ensemblealg isa EnsembleGPUArrayAutonomous
+        ps = CuArray([SVector{length(probs[i].p)}(probs[i].p) for i in 1:length(I)])
+        if typeof(alg) <: OrdinaryDiffEq.Tsit5
+            #Adaptive version only works with saveat
+            if haskey(kwargs, :saveat)
+                saveat = kwargs[:saveat]
+                solus = vectorized_asolve(ensembleprob.prob, ps, GPUSimpleATsit5(); saveat)
+            else
+                solus = vectorized_solve(ensembleprob.prob, ps, GPUSimpleTsit5())
+            end
+            _callback = generate_callback(probs[1], length(I), ensemblealg; kwargs...)
+            sol = solve(ensembleprob.prob, alg; kwargs..., callback=_callback, merge_callbacks=false,
+                internalnorm=diffeqgpunorm)
+            sol, solus
+        else
+            error("We don't have solvers implemented for this algorithm yet")
+        end
+    elseif ensemblealg isa EnsembleGPUArray
+        batch_solve_up(ensembleprob, probs, alg, ensemblealg, I, u0, p; kwargs...)
+    end
     [ensembleprob.output_func(SciMLBase.build_solution(probs[i],alg,sol.t,solus[i],destats=sol.destats,retcode=sol.retcode),i)[1] for i in 1:length(probs)]
```

Comment:
This code performs scalar indexing on GPU Arrays when it solves from `EnsembleGPUArrayAutonomous`. What to do in this case?

Reply:
What is scalar indexing on a GPU array here? I don't see where that would happen.

Reply:
`solus[i]` is the scalar indexing.

Reply:
but that's an array of solutions?